jpa/subsystem/src/main/java/org/jboss/as/jpa/service/PersistenceUnitServiceImpl.java

```diff
@@ -169,11 +175,21 @@ public Void run() {
                                         if (beanManagerInjector.getOptionalValue() != null) {
                                             proxyBeanManager = new ProxyBeanManager();
                                             proxyBeanManager.setDelegate(beanManagerInjector.getOptionalValue());
-                                            properties.getValue().put(CDI_BEAN_MANAGER, proxyBeanManager);
+                                            wrapperBeanManagerLifeCycle = persistenceProviderAdaptor.beanManagerLifeCycle(proxyBeanManager);
```

Comment:
There are a few things going on here.  If phaseOnePUS is non-null, we are handling a 2 phase bootstrap approach, which ensures that JPA starts before there is a CDI beanmanager, so that JPA can rewrite the application classes before Weld (eagerly) loads the application classes.  

In the above code, phaseOnePersistenceUnitService is null, meaning that we are bootstrapping JPA via a single phase.  If it helps, I could update the pull request to add comments.